qiskit/transpiler/passes/mapping/check_cnot_direction.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+This pass checks if the CNOTs (or any other 2Q) in the DAG follow the right
+direction with respect to thecoupling map.
+"""
+
+from qiskit.transpiler._basepasses import AnalysisPass
+from qiskit.mapper import Layout
+from qiskit.extensions.standard.swap import SwapGate
+
+
+class CheckCnotDirection(AnalysisPass):
+    """
+    checks if the CNOTs (or any other 2Q) in the DAG follow the right
+    direction with respect to thecoupling map
+    """
+
+    def __init__(self, coupling_map, initial_layout=None):
+        """
+        Checks if the 2Q gates in DAGCircuit are in the allowed direction with
+        respect to `coupling_map`.
+        Args:
+            coupling_map (CouplingMap): Directed graph represented a coupling map.
+            initial_layout (Layout): The initial layout of the DAG to analyze.
+        """
+        super().__init__()
+        self.layout = initial_layout
+        self.coupling_map = coupling_map
+
```

Comment:
Should `CheckCnotDirection` require `CheckMap`? I'm on the fence since, practically, `CheckCnotDirection` can be run without having run `CheckMap`, but logically, a dag can't be `is_direction_mapped` without also first being `is_swap_mapped`.

Reply:
I think we can set `requires` later, with a clear workflow. It's hard for me to understand how are going to be orchestrated. You are raising a very valid point here..